docs: unify documentation workflows with improved validation

EdwardAngert opened the pull request:
This PR is a replacement for #17522 with a branch name that works better with our docs preview system.

## Summary
- Combine all documentation validation workflows into a unified approach
- Fix errors in lychee-action configuration for link checking
- Remove duplicate example workflows
- Fix caching issues in build process
- Update branch name format for docs preview URLs

## Test plan
- Verify all documentation workflows run without errors
- Test docs preview URLs work correctly
- Ensure link checking functions properly

Note: This branch replaces feature/unified-docs-workflow-combined to resolve docs preview URL issues.

matifali commented on .github/actions/setup-node/action.yaml:
```diff
@@ -19,7 +19,9 @@ runs:
         cache: "pnpm"
-        cache-dependency-path: ${{ inputs.directory }}/pnpm-lock.yaml
+        cache-dependency-path: |
+          pnpm-lock.yaml
+          ${{ inputs.directory }}/pnpm-lock.yaml
```
This is an unrelated change.

matifali commented on .github/docs/actions/docs-preview/action.yaml:
How many times do these actions get reused? If it's only a single time, we should probably not create an action and put it directly in the workflow. 

There is benefit that it makes the workflow file look simple. So can keep it too :)

EdwardAngert replied:
claude and I are currently having a late-night argument about simplifying all of this 😫 

I put it all into a single PR to try to streamline it - but I'm definitely going to have to continue this in the morning

matifali commented on .github/docs/actions/docs-setup/action.yaml:
```diff
@@ -0,0 +1,47 @@
+  using: 'composite'
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
```
Can you use commit hashes for all action versions? See `ci.yaml`

matifali commented on .github/docs/vale/styles/GitLab/Spelling.yml:
```diff
@@ -0,0 +1,37 @@
+  "sign[ -]in": sign in
+  "ssh key": SSH key
+  "two factor": two-factor
+  "web ?hook": webhook
```
I would also add `jetbrains: JetBrains` and other partner or integration vendor names in the correct case. Like `jfrog: JFrog`

matifali commented on .github/workflows/docs-ci.yaml:
```diff
@@ -28,7 +28,7 @@ jobs:
         uses: ./.github/actions/setup-node
 
-      - uses: tj-actions/changed-files@9934ab3fdf63239da75d9e0fbd339c48620c72c4 # v45.0.7
+      - uses: tj-actions/changed-files@v45
```
```suggestion
      - uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46.0.5
```
Let's keep using the commit hashes.

matifali commented on .github/workflows/docs-link-check.yaml:
```diff
@@ -0,0 +1,193 @@
+          fi
+
+      # Create a single GitHub issue for all problems found
+      - name: Create GitHub issue
+        if: steps.process-results.outputs.has_issues == 'true'
+        id: create-issue
+        uses: peter-evans/create-issue-from-file@v4.0.1
+        with:
+          title: ${{ steps.process-results.outputs.issue_title }}
+          content: ${{ steps.process-results.outputs.issue_content }}
+          labels: |
+            documentation
+            bug
+            needs-triage
+          # Configure assignee through CODEOWNERS or a team instead of hardcoding
```
I don't think it's a good idea to create issues from a workflow. They can grow pretty quickly, and we also have flakes.

matifali reviewed (approved):
I am ok with this. But we need to fix all action version to latest commit hashes see how its done in `ci.yaml`

The PR looks too big. SO would prefer if we reduce the size and remove features we are not using. I ave seen for docs-shared action that many inputs were set to `false` when used. So it would be nice to remove any features from the actions we do not use anywhere.

EdwardAngert committed:
refactor: implement docs workflow with MegaLinter and standardized actions

- Replace custom implementation with MegaLinter documentation flavor
- Use tj-actions/changed-files for file detection
- Create standard configuration files for all tools
- Remove complex composite action in favor of pipeline approach
- Update documentation to reflect new architecture
- Add testing script for validation
- Fix Vale style checking implementation
- Cleanup documentation files

EdwardAngert committed:
fix: correct variable syntax in workflow YAML

EdwardAngert committed:
fix: resolve YAML syntax issues and add Vale configuration

- Fixed YAML syntax in docs-unified.yaml workflow
- Added Vale configuration with basic style rules
- Created template file approach for PR comment to avoid syntax issues
- Ensured proper Vale directory structure

EdwardAngert committed:
style: improve YAML formatting in docs workflow

- Added YAML document start marker
- Removed trailing whitespace
- Added newline at end of file

EdwardAngert committed:
fix: add Node.js setup to docs-unified workflow

- Added Node.js setup step in the workflow to resolve errors with pnpm installation
- Uses existing ./.github/actions/setup-node action to maintain consistency with other workflows

EdwardAngert committed:
fix: secure GitHub Actions workflow to pass actionlint

- Follow security best practice for GitHub Actions
- Pass potentially untrusted context values via environment variables
- Fixes actionlint warning about using github.head_ref directly in scripts

EdwardAngert committed:
feat: enhance PR comments with timestamps and progress updates

- Add timestamps to initial, in-progress, and completion comments
- Add a comment update after file detection to show progress
- Improve user experience with better status messaging
- Allow tracking of workflow progress through multiple stages

EdwardAngert committed:
fix: use platform-compatible date formatting in workflow

- Fix date formatting to be compatible with GitHub Actions runners (Linux)
- Use TZ=UTC instead of -u flag for better compatibility
- Remove unnecessary timestamp environment variable
- Ensure all date commands use the same format

EdwardAngert committed:
feat: simplify PR comment system in docs workflow

- Use peter-evans/create-or-update-comment action for PR comments
- Replace marocchino/sticky-pull-request-comment with established pattern
- Remove complex date formatting and temporary file management
- Add find-comment action to track PR comments consistently
- Follow existing workflow patterns from PR deploy workflow
